src/preprocessing/passes/bv_to_int.cpp

```diff
@@ -269,560 +241,541 @@ Node BVToInt::bvToInt(Node n)
 {
   n = eliminationPass(n);
   n = makeBinary(n);
-  vector<Node> toVisit;
-  toVisit.push_back(n);
   uint64_t granularity = options::solveBVAsInt();
 
-  while (!toVisit.empty())
+  for (Node current : NodeDfsIterable(n).inPostorder())
   {
-    Node current = toVisit.back();
     uint64_t currentNumChildren = current.getNumChildren();
-    if (d_bvToIntCache.find(current) == d_bvToIntCache.end())
+    if (d_bvToIntCache.find(current) != d_bvToIntCache.end())
     {
-      // This is the first time we visit this node and it is not in the cache.
-      d_bvToIntCache[current] = Node();
```

Comment:
@yoni206 why were you doing this?
And how do the changes capture that?

Reply:
In this kind of traversal, there are three states for nodes:

1. Not yet previsited
2. Previsited (children added) by not post-visited
3. Postvisited

I believe state (1) was represented as "not in the cache", (2) was represented as "in the cache, mapped to Null", and (3) was represented as "in the cache, mapped to something non-null".

The post-order iterator handles all of this for you. Now the cache just stores the mapping from nodes to what they're processed into.